File: console/src/features/points/dto/point-transaction-response.ts
export type TransactionType = "Award" | "Deduct" | "Transfer" | "Exchange";

export interface PointTransactionResponse {
  project_id: string;
  meta_user_id: string;
  month: string;
  transaction_type: TransactionType;
  amount: number;
  target_user_id?: string;
  description?: string;
  created_at: number;
}

export interface ListPointTransactionsResponse {
  items: PointTransactionResponse[];
  bookmark?: string;
}

File: app/src/features/points/dto/point_transaction_response.rs
use serde::{Deserialize, Serialize};
use crate::common::types::Partition;
use crate::features::points::TransactionType;

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct PointTransactionResponse {
    pub project_id: Partition,
    pub meta_user_id: String,
    pub month: String,
    pub transaction_type: TransactionType,
    pub amount: i64,
    pub target_user_id: Option<String>,
    pub description: Option<String>,
    pub created_at: i64,
}
